File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/monitoring/port/MemoryExtractionMetricsPort.java
package com.study.webflux.rag.application.monitoring.port;

public interface MemoryExtractionMetricsPort {
	void recordExtractionTriggered();

	void recordExtractionSuccess(int count);

	void recordExtractionFailure();

	void recordExtractedMemoryType(String type, int count);

	void recordExtractedImportance(double importance);
}

File: monitoring/src/main/kotlin/com/miyou/app/monitoring/port/MemoryExtractionMetricsPort.kt
package com.miyou.app.monitoring.port

interface MemoryExtractionMetricsPort {
    fun recordExtractionTriggered()

    fun recordExtractionSuccess(count: Int)

    fun recordExtractionFailure()

    fun recordExtractedMemoryType(
        type: String,
        count: Int,
    )

    fun recordExtractedImportance(importance: Double)
}
